### DISCUSSION TITLE
Use govc datastore.download for entire directory

### DISCUSSION THREAD

**Discussion by aterrell5** [vmware/govmomi#3196] [Community Help] [answered]
Is it possible to use govc to download files in batches from a datastore? After looking at the documentation and trying a few methods, I've been unable to find a way to script the collection of all files/resources within a directory on a datastore. Is this possible? Here is what I've been trying (in various iterations including wildcards, etc): 

Dir Structure: 
DATASTORE_NAME
|_ path on datastore
  |_ sub path
    |_ sub sub path
    |_  files here
  |_ sub sub path
    |_ files here
  |_ files here
|_ files here

govc datastore.download -ds="$DATASTORE_NAME" "/path on datastore/sub path/" "/localOS/storage/sub path/" (I've also tried like /sub path/* and similar to see if it was just a syntax issue)

This appears to start the process but errors out shortly after saying:
govc: open /localOS/storage/sub path/: is a directory 

Any ideas what I'm missing or can someone point me in the right direction?

**Comment by dougm** [comment, marked as answer]
No, datastore.download only copies a single file. Can you describe your use-case? There is govc export.ovf, which can download a VM and its related files.

**Reply by aterrell5** [reply]
I see and thanks for the info. I think making use of powerCLI and new-psdrive pointing to the datastore might be our alternative. We have a datastore that holds ISOs and Application packages for a logically air-gapped environment. Was hoping to build out a solution to script the retrieval of all the components from that datastore once connected.
